osf/management/commands/migrate_registration_responses.py

```diff
@@ -0,0 +1,122 @@
+import datetime
+import logging
+
+from django.core.management.base import BaseCommand
+from django.apps import apps
+from tqdm import tqdm
+
+from bulk_update.helper import bulk_update
+from framework.celery_tasks import app as celery_app
+
+logger = logging.getLogger(__name__)
+
+def migrate_registrations(dry_run, rows='all'):
+    """
+    Loops through registrations whose registration_responses have not been migrated,
+    and pulls this information from the "registered_meta" and flattens it, with
+    keys being the "registration_response_key"s and values being the most deeply
+    nested user response in registered_meta
+    """
+    AbstractNode = apps.get_model('osf.AbstractNode')
+    registrations = AbstractNode.objects.filter(
+        registration_responses_migrated=False,
+        type='osf.registration'
+    )
+    regs_count = len(registrations)
```

Comment:
gah, thanks

Reply:
hmm, wait, I think I did this because I'm looping through every resource anyway in `migrate_resources`.  since I have to loop through the queryset, calling len() caches the results first, so I don't have to make another db request.

Reply:
In that case, you'll want to slice `registrations[:rows]` first, so it's not caching *every* unmigrated registration, just the ones needed this time.

Reply:
good point!

Reply:
ahh yeah, that makes sense 👍